### ISSUE TITLE
foreground and background maps in appendix

### ISSUE THREAD

**Issue by XIAO1HAI** [dahyun-kang/ifsl#6]
![image](https://user-images.githubusercontent.com/90389233/197729756-507f1f4c-0bdf-4afa-b872-c45c02c9b439.png)
· Hello, I would like to ask you that the foreground map in the figure above in the appendix is realized by converting the original img image to grayscale image and then using a threshold of 0.5 to realize the foreground and background map. Is there any additional processing code？ Only the following code blocks related to the foreground background map were found in the paper.
![image](https://user-images.githubusercontent.com/90389233/197731015-8340d4af-c5de-4fdc-b043-57494fab1dda.png)
![image](https://user-images.githubusercontent.com/90389233/197731379-3827bc1d-1702-4c37-8359-272cf771dcf3.png)

**Comment by dahyun-kang** [comment]
Hello again,


After the `shared_fg_masks` passes `merge_bg_mask()`, the visualization is from `logit_mask[:, 0]` of which 0-th channel is the episodic background channel.

Have a great day! 😃

**Comment by XIAO1HAI** [comment]
Haha, I see. It suddenly dawned on me

But it should be like this visualization is from `class_activation = logit_mask[:, :, 0].max(dim=-1)[0].max(dim=-1)[0] >= math.log(0.5)` .`logit_mask[:, :, 0]` instead of `logit_mask[:, 0]`？

**Comment by dahyun-kang** [comment]
Ah, I didn't do any max pooling operation for visualizing the background channel. You can simply try visualizing `logit_mask[:, 0]` and see how it looks. `collect_class_presence` is for classification evaluation.

**Comment by XIAO1HAI** [comment]
OK, I will try to do it. Thank you for your warm reply ！best wishes

**Comment by XIAO1HAI** [comment]
OK, I will try to do it. Thank you for your warm reply ！best wishes

**Comment by XIAO1HAI** [comment]
> How can I modify it to achieve the visual effect in your article? I have tried many times but failed to achieve it. I hope you can give me some advice

**Comment by XIAO1HAI** [comment]
> Hello again,
> 
> After the `shared_fg_masks` passes `merge_bg_mask()`, the visualization is from `logit_mask[:, 0]` of which 0-th channel is the episodic background channel.
> 
> Have a great day! 😃

How can I modify it to achieve the visual effect in your article? I have tried many times but failed to achieve it. I hope you can give me some advice

**Comment by dahyun-kang** [comment]
I don't have the visualization code left, but it was really simple such as below for instance:

```python
import matplotlib.pyplot as plt
plt.imshow(logit_mask[0, 0].cpu().numpy())  # visualizing the background channel of 0-th instance in a batch
```

It wouldn't be as nice as my examples for some examples, but you'll find some nicely segmented backgrounds if you try more.
Plus, please make sure whatever you're visualizing is between 0 and 1 as my implementation uses log-softmax to prevent numerical unstability.

**Comment by XIAO1HAI** [comment]
OK, thank you. Thank you very much for your suggestion.

 I have always wanted to make changes in vis.py to achieve vision. I will try again. 

Thank you. I wish you all the best

**Comment by dahyun-kang** [comment]
Thank you, you too!
Closing the issue.
